proto/backup.proto

```diff
@@ -187,6 +188,23 @@ message GCS {
     string credentials_blob = 6;
 }
 
+// S3 and GCS can be transitioned to use this
+message Bucket {
+    string endpoint = 1;
+    string region = 3;
+    string bucket = 4;
+    string prefix = 5;
+    string storage_class = 6;
+}
+
+// CloudDynamic allows adding new fields without changing protobuf definitions
+message CloudDynamic {
+    Bucket bucket = 1;
+    string provider_name = 2;
```

Comment:
I know we cannot define constant in protobuf. But can this be an enum? or at least inline comment what are the possible provider names.

Reply:
I can comment that s3/gcs are available. However the entire point of this is that it is dynamic and that we don't have to alter all of our code paths to experiment with new cloud providers.

Reply:
I'm fine just comment what are the possible values. But I think the purpose of protobuf is to make things well-defined. Otherwise we could use JSON or XML just for flexibility.

Reply:
Protobuf supports dynamic maps and also arrays of dynamic length via repeat. XML has always supported strong schemas and these are available for JSON as well now. I think the main reason for Protobuf is a more efficient encode/decode.
Most of the time things should be well-defined. Occasionally we want things to be more dynamic. Just as usually we are happy with a strong schema in a database but now we support JSON as well for the times where it is useful.